lib/plugins/nwc/src/model.rs

```diff
@@ -0,0 +1,118 @@
+use nostr_sdk::{nips::nip47::NostrWalletConnectURI, PublicKey};
+use serde::{Deserialize, Serialize};
+
+use crate::DEFAULT_RELAY_URLS;
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct NwcConfig {
+    /// A list of default relay urls to add per connection
+    pub relay_urls: Option<Vec<String>>,
+    /// Custom Nostr secret key (hex-encoded) for the wallet node
+    pub secret_key_hex: Option<String>,
+}
+
+impl NwcConfig {
+    pub fn relays(&self) -> Vec<String> {
+        self.relay_urls
+            .clone()
+            .unwrap_or(DEFAULT_RELAY_URLS.iter().map(|s| s.to_string()).collect())
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct PeriodicBudget {
+    /// The amount of budget used (in satoshi) for the period
+    /// Resets once every period ([PeriodicBudget::reset_time_sec])
+    pub used_budget_sat: u64,
+    /// The maximum budget amount allowed (in satoshi) for the period
+    pub max_budget_sat: u64,
+    /// The duration of the budget's period
+    /// ## Dev Note:
+    /// If the reset time is less than [crate::MIN_REFRESH_INTERVAL_SEC] seconds,
+    /// then it will take at most [crate::MIN_REFRESH_INTERVAL_SEC] seconds in order for the
+    /// reset to take effect.
+    pub reset_time_sec: u32,
+    /// The latest budget update time (last reset time)
+    pub updated_at: u32,
+}
+
+impl PeriodicBudget {
+    pub(crate) fn from_budget_request(req: PeriodicBudgetRequest, updated_at: u32) -> Self {
+        Self {
+            used_budget_sat: 0,
+            max_budget_sat: req.max_budget_sat,
+            reset_time_sec: req.reset_time_sec,
+            updated_at,
+        }
+    }
+}
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct PeriodicBudgetRequest {
+    /// See [PeriodicBudget::max_budget_sat]
+    pub max_budget_sat: u64,
+    /// See [PeriodicBudget::reset_time_sec]
+    pub reset_time_sec: u32,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct NwcConnection {
+    /// The NWC uri for the connection
+    pub connection_string: String,
+    /// The creation time of the connection
+    pub created_at: u32,
+    /// Specifies whether this is a receive-only connection. Defaults to false.
+    pub receive_only: bool,
+    /// The expiry time of the connection
+    /// ## Dev Note:
+    /// If the expiry time is less than [crate::MIN_REFRESH_INTERVAL_SEC] seconds,
+    /// then it will take at most [crate::MIN_REFRESH_INTERVAL_SEC] seconds in order for the
+    /// connection to be deleted.
+    pub expiry_time_sec: Option<u32>,
```

Comment:
Shouldn't we expose `expiration_timestamp` instead? How does this translate to the end-user? Do they have to calculate expiration time by adding up `created_at` & `expiry_time_sec`?